rs/registry/canister/src/mutations/node_management/do_remove_node_directly.rs

```diff
@@ -121,21 +121,30 @@ impl Registry {
             );
         }
 
-        // 3. Ensure the node is not an API Boundary Node.
-        // In order to succeed, a corresponding ApiBoundaryNodeRecord should be removed first via proposal.
-        let api_bn_id = self.get_api_boundary_node_record(payload.node_id);
-        if api_bn_id.is_some() {
-            panic!(
-                "{}do_remove_node_directly: Cannot remove a node, as it has ApiBoundaryNodeRecord with record_key: {}",
-                LOG_PREFIX,
-                make_api_boundary_node_record_key(payload.node_id)
-            );
+        let mut mutations = vec![];
+
+        // 3. Check if the node is an API Boundary Node. If there is a replacement node, remove the existing node and try to assign the new one to act as API boundary node. This will only .
+        if let Some(api_bn_record) = self.get_api_boundary_node_record(payload.node_id) {
+            if let Some(node_id) = new_node_id {
```

Comment:
How about

```
let Some(new_node_id) = new_node_id else {
  old behavior
};

new behavior
```

? Admittedly, this is debatable, and not a big difference, but I think it's an improvement.